CMakeLists.txt

```diff
@@ -7,85 +7,103 @@ set(EIPSCANNER_FULL_VERSION ${EIPSCANNER_MAJOR_VERSION}.${EIPSCANNER_MINOR_VERSI
 
 project(eip_scanner VERSION ${EIPSCANNER_FULL_VERSION})
 
+find_package(catkin REQUIRED)
 
-set(CMAKE_CXX_STANDARD 14)
-option(ENABLE_VENDOR_SRC "Enable vendor source" OFF)
-
-find_package(catkin REQUIRED COMPONENTS
-)
-
-
-include_directories(${CMAKE_SOURCE_DIR}/src)
-
-set(SOURCE_FILES
-        src/cip/connectionManager/ForwardCloseRequest.cpp
-        src/cip/connectionManager/ForwardOpenRequest.cpp
-        src/cip/connectionManager/LargeForwardOpenRequest.cpp
-        src/cip/connectionManager/ForwardOpenResponse.cpp
-        src/cip/connectionManager/NetworkConnectionParametersBuilder.cpp
-
-        src/cip/CipRevision.cpp
-        src/cip/EPath.cpp
-        src/cip/MessageRouterRequest.cpp
-        src/cip/MessageRouterResponse.cpp
-        src/cip/CipString.cpp
-
-        src/eip/CommonPacket.cpp
-        src/eip/CommonPacketItem.cpp
-        src/eip/CommonPacketItemFactory.cpp
-        src/eip/EncapsPacket.cpp
-        src/eip/EncapsPacketFactory.cpp
-
-        src/fileObject/FileObjectEmptyState.cpp
-        src/fileObject/FileObjectLoadedState.cpp
-        src/fileObject/FileObjectNonExistentState.cpp
-        src/fileObject/FileObjectState.cpp
-        src/fileObject/FileObjectUploadInProgressState.cpp
-
-        src/sockets/BaseSocket.cpp
-        src/sockets/EndPoint.cpp
-        src/sockets/TCPSocket.cpp
-        src/sockets/UDPBoundSocket.cpp
-        src/sockets/UDPSocket.cpp
-
-        src/utils/Logger.cpp
-        src/utils/Buffer.cpp
-
-        src/BaseObject.cpp
-        src/ConnectionManager.cpp
-        src/DiscoveryManager.cpp
-        src/FileObject.cpp
-        src/IdentityObject.cpp
-        src/IOConnection.cpp
-        src/MessageRouter.cpp
-        src/ParameterObject.cpp
-        src/SessionInfo.cpp)
-
+set(EIP_SOURCE_DIR ${CMAKE_SOURCE_DIR}/src)
+set(EIP_INCLUDE_DIR ${CATKIN_DEVEL_PREFIX}/include)
+file(MAKE_DIRECTORY ${EIP_INCLUDE_DIR})
 
 catkin_package(
-        INCLUDE_DIRS src
-        LIBRARIES eip_scanner
-
-  #      DEPENDS EIGEN3
+    INCLUDE_DIRS ${EIP_INCLUDE_DIR} ${EIP_INCLUDE_DIR}/eip_scanner
```

Comment:
why both "${EIP_INCLUDE_DIR}" and "${EIP_INCLUDE_DIR}/eip_scanner"?

Reply:
Because the includes inside the headers are absolute and expect the files in the top level include directory. If the headers are included in other projects, then their includes won't be found.
This was the easiest solution that does not imply changing all the headers.